### PR TITLE
Add support for dynamic forms in helm install and upgrade

### PR DESCRIPTION
**Pull request by rohitkrai03**
**Fixes**: 
- https://issues.redhat.com/browse/ODC-4083
- https://issues.redhat.com/browse/ODC-3061

**Depends on**: https://github.com/openshift/console/pull/5789

**Analysis / Root cause**: We have YAML editor as default editor for values in Helm install and upgrade form. We need to support dynamic form generation if the helm chart has a value.schema.json present. 

**Solution Description**: Once we have the custom formik fields to handle dynamic forms and toggle between form and yaml editor without loosing context, we need to use that to add support for synced editor in helm install and upgrade form.

**Screen shots / Gifs for design review**: 
Helm Install - 
Dynamic Forms Helm Install

Helm Upgrade - 
Dynamic Forms Helm Upgrade

**Browser conformance**: 

- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

**Comment by rohitkrai03** [comment]
/hold

This PR depends on components being added in https://github.com/openshift/console/pull/5789.

**Comment by christianvogt** [comment]
I get this error on all helm charts:
image

**Comment by christianvogt** [comment]
@rohitkrai03 is it possible to keep the expanded state of the form view when switching between form or yaml?

**Comment by rohitkrai03** [comment]
> @rohitkrai03 is it possible to keep the expanded state of the form view when switching between form or yaml?

@christianvogt Don't think it's possible to maintain that state, it's handled by `DynamicForm` component and the templates that it's using. I'll look into it and see if somehow we can do something.

**Comment by andrewballantyne** [comment]
@rohitkrai03 This may be something lost here... The "Root Schema" feels weird:

image

Would we not want to auto-expand the first level of the schema to avoid the collapsed block?

**Comment by rohitkrai03** [comment]
> @rohitkrai03 There may be something lost here... The "Root Schema" feels weird:
> 
> image
> 
> Would we not want to auto-expand the first level of the schema to avoid the collapsed block?

@andrewballantyne That actually depends on the schema provided by the chart. If the root schema has a `title` and `description`, the form will render it inside an `ExpandCollapse`. The current schema in the example Helm chart is not written properly and has some schema errors. I plan to raise a PR to update that schema in Helm catalog. The gif above is using the custom schema that I created and not the schema that is coming from API. That's why you're seeing that difference.

**Comment by christianvogt** [comment]
Overall it seems to work well.

Very annoying that the form collapses when switching from yaml to form; discarding the previous expanded states. I think if this merges as is we should raise an issue to capture this request and discuss with UX.

cc @openshift/team-devconsole-ux

**Comment by serenamarie125** [comment]
This is looking great @rohitkrai03 !  Agree with the comment made by @christianvogt around remember expansion.  Is this being handled differently for the operator backed services/operator OLM based forms?

**Comment by rohitkrai03** [comment]
> This is looking great @rohitkrai03 ! Agree with the comment made by @christianvogt around remember expansion. Is this being handled differently for the operator backed services/operator OLM based forms?

@serenamarie125 Even OLM forms don't have a way to keep the expansion state while switching between form and yaml views. I can create a issue to look into this later.

**Comment by rohitkrai03** [comment]
@christianvogt @serenamarie125 Bug to track the issue - https://issues.redhat.com/browse/ODC-4228

File: frontend/packages/dev-console/src/components/helm/HelmInstallUpgradePage.tsx [hunk 1]
```diff
@@ -1,5 +1,7 @@
 import * as React from 'react';
 import * as _ from 'lodash';
+import * as Ajv from 'ajv';
+import { JSONSchema6 } from 'json-schema';
```
Comment (christianvogt) [inline frontend/packages/dev-console/src/components/helm/HelmInstallUpgradePage.tsx:4]: We should have dependencies on these these 2 packages in our `package.json`

Comment (rohitkrai03) [reply frontend/packages/dev-console/src/components/helm/HelmInstallUpgradePage.tsx:4]: `json-schema` is being used just for its `JSONSchema6` type definition all over the console. Is it necessary to add it to `package.json` if only type definition is being used?

Comment (rohitkrai03) [reply frontend/packages/dev-console/src/components/helm/HelmInstallUpgradePage.tsx:4]: `react-jsonschema-form` has a dependency on `ajv` but it's probably better to make it a direct dependency.

Comment (rohitkrai03) [reply frontend/packages/dev-console/src/components/helm/HelmInstallUpgradePage.tsx:4]: Updated `package.json` with new dependencies.

**Pull request merged**